packages/mdx/lib/plugin/rehype-remove-raw.js

```diff
@@ -1,5 +1,6 @@
 /**
  * @typedef {import('hast').Root} Root
+ * @typedef {import('mdast-util-to-hast').Raw} DoNotRemoveUsedToAddRawToNodeType
```

Comment:
@wooorm @remcohaszing do you have a trick for getting the raw node type to register itself?
I tried this, it doesn't seem to work.
The code below needs to be able to filter for the `raw` node type (a custom node added by this plugin) from a `hast` tree

Reply:
Maybe it registers without the `.Raw`? (https://github.com/syntax-tree/mdast-util-to-hast#types)

But I thought this does register it?

Reply:
`mdast-util-to-hast` needs to be loaded. THis can be done by either:

1. Using a TypeScript tripple slash directive (not sure if it works in JSDoc too)
   ```ts
   /// <reference types="mdast-util-to-hast" />
   ```
   This means the these types will also be loaded for people who use this file
2. Adding it to `"types"` in `tsconfig.json` compiler options.
   ```jsonc
   {
     // …
     "compilerOptions": {
       // …
       "types": ["mdast-util-to-hast"]
     }
   }
   ```

I recommend using the latter.

Regardless of this issue, I recommend using the `"types"` compiler option in a mono repo like this. It prevents TypeScript from loading unwanted type definitions from `@types` dependencies. E.g. without it, the TypeScript loads `@types/react` while checking `@mdx-js/preact` and `@mdx-js/vue`.

Reply:
Thanks for the suggestions!

🤔 I've tried adding

```json
  "compilerOptions": {
    "types": ["mdast-util-to-hast", "@types/node"]
  }
```
to _tsconfig.json_

as well as

```js
/**
 * @typedef {import('mdast-util-to-hast')} DoNotRemoveUsedToAddRawToNodeType
 */
```

both still show `raw` type as `never` 🤔 

---

Will keep trying different options.

Reply:
committed in bd4c814fe7b9ad756e29795aad4fce48f540ed53 for reference

Reply:
The problem is that `mdast-util-to-hast` was compiled before TS 4.9.
It uses `export type Raw = import('./complex-types').Raw` in there, which is broken, because there is no extension.
`mdast-util-to-hast` has now been published with types that were compiled with TS 4.9.
If you update the package lock (and rebase), this PR will work!

Reply:
Updated in [746afe3](https://github.com/mdx-js/mdx/pull/2228/commits/746afe3d07199af0d28b761da5bc572808096c83)